### PR TITLE
Remove unnecessary form on order success page

### PR DESCRIPTION
**Pull request by danielgoodwin97**
### Description (*)
A form with an input which does the same job as an anchor tag.

### Manual testing scenarios (*)
1. Add Item to basket
2. Checkout as guest
3. Successfully pay
4. Check 'Create an Account' button

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=22126) <br/>All committers have signed the CLA.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-4649](https://jira.corp.magento.com/browse/ENGCOM-4649) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@danielgoodwin97 thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by VasylShvorak** [comment]
:heavy_check_mark: QA passed

File: app/code/Magento/Checkout/Test/Mftf/Section/CheckoutSuccessMainSection.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
         <element name="orderLink" type="text" selector="a[href*=order_id].order-number" timeout="30"/>
         <element name="orderNumberText" type="text" selector=".checkout-success > p:nth-child(1)"/>
         <element name="continueShoppingButton" type="button" selector=".action.primary.continue" timeout="30"/>
-        <element name="createAnAccount" type="button" selector="input[value='Create an Account']" timeout="30"/>
+        <element name="createAnAccount" type="button" selector="#registration a.action.primary span[data-bind*=&quot;i18n: 'Create an Account'&quot;]" timeout="30"/>
```
Comment (orlangur) [inline app/code/Magento/Checkout/Test/Mftf/Section/CheckoutSuccessMainSection.xml]: @p-bystritsky please modify all selectors to satisfy https://devdocs.magento.com/guides/v2.3/coding-standards/code-standard-demarcation.html

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Checkout/Test/Mftf/Section/CheckoutSuccessMainSection.xml (hunk 1 above)
Comment (p-bystritsky) [reply app/code/Magento/Checkout/Test/Mftf/Section/CheckoutSuccessMainSection.xml]: @orlangur done.

Comment (orlangur) [reply app/code/Magento/Checkout/Test/Mftf/Section/CheckoutSuccessMainSection.xml]: @p-bystritsky I don't see... In functional tests data attributes, names etc must be used. No IDs, CSS classes and such allowed.

Comment (p-bystritsky) [reply app/code/Magento/Checkout/Test/Mftf/Section/CheckoutSuccessMainSection.xml]: @orlangur updated.

File: app/code/Magento/Checkout/Test/Mftf/Section/CheckoutSuccessMainSection.xml [hunk 2]
```diff
@@ -16,7 +16,7 @@
         <element name="orderLink" type="text" selector="a[href*=order_id].order-number" timeout="30"/>
         <element name="orderNumberText" type="text" selector=".checkout-success > p:nth-child(1)"/>
         <element name="continueShoppingButton" type="button" selector=".action.primary.continue" timeout="30"/>
-        <element name="createAnAccount" type="button" selector="input[value='Create an Account']" timeout="30"/>
+        <element name="createAnAccount" type="button" selector="[data-bind*=&quot;i18n: 'Create an Account'&quot;]" timeout="30"/>
```
Comment (orlangur) [inline app/code/Magento/Checkout/Test/Mftf/Section/CheckoutSuccessMainSection.xml:19]: @p-bystritsky note that binding by `[value]` or `[name]` is just fine too.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-4649](https://jira.corp.magento.com/browse/ENGCOM-4649) has been created to process this Pull Request

**Comment by VasylShvorak** [comment]
:heavy_check_mark: QA passed

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Magento\FunctionalTestingFramework.functional.CreateNewAccountAfterCheckoutTest is failing on this pull request with the following message:

    Link or Button or CSS or XPath element with 'input.action.primary' was not found.

@p-bystritsky could you please help to resolve this fail

**Comment by p-bystritsky** [comment]
@sivaschenko done

**Comment by sidolov** [comment]
Hi @p-bystritsky , could you please take a look at failed test: 
`Magento\FunctionalTestingFramework.functional.CreateNewAccountAfterCheckoutTest`
Failed with error: `Field by name, label, CSS or XPath element with '#password' was not found.`

**Comment by p-bystritsky** [comment]
@sidolov fixed.

**Pull request merged**